response.go

```diff
@@ -961,6 +992,43 @@ func (r *Response) getJSONP(
 	return value
 }
 
+// Reader returns the body reader from the response.
+//
+// Reader replaces the other methods for reading response body
+// and disables rewinding when reading.
```

Comment:
User does not know about rewinding actually, it's our implementation details.

Let's say something like:

> Reader is mutually exclusive with method that read the whole response body, like Text, Body, JSON, etc